torch/_functorch/aot_autograd.py

```diff
@@ -3042,7 +3044,10 @@ def functional_call(*args, **kwargs):
         # can now be done safely. (2) Dynamo logic protects the 1:1 sizing below.
         for name in params.keys():
             assert name in mod._param_name_to_source, f"{name} not found."
-            aot_autograd_arg_pos_to_source.append(mod._param_name_to_source[name])
+            source = mod._param_name_to_source[name]
+            assert source not in seen_sources, source
```

Comment:
nit: better error message?

Reply:
i plead "this is an assert error we shouldn't hit it ever lol"

Reply:
If you're confident this will never happen, I'm good with it!